zivid_camera/src/zivid_camera.cpp

```diff
@@ -58,6 +58,36 @@ std::string toString(zivid_camera::CameraStatus camera_status)
   return "N/A";
 }
 
+template <typename ConfigDRServerType, typename ZividSettings>
+std::unique_ptr<ConfigDRServerType> setupConfigDRServer(const std::string& serverName, ros::NodeHandle& nh,
+                                                        const ZividSettings& defaultSettings)
+{
+  auto server = std::make_unique<ConfigDRServerType>(serverName, nh);
+
+  // Setup min/max/default and current config based on the provided defaultSettings
+  using ConfigType = typename ConfigDRServerType::ConfigType;
+  const auto min_config = getMinConfigFromZividSettings<ConfigType>(defaultSettings);
+  server->dr_server.setConfigMin(min_config);
+
+  const auto max_config = getMaxConfigFromZividSettings<ConfigType>(defaultSettings);
+  server->dr_server.setConfigMax(max_config);
+
+  const auto default_config = getDefaultConfigFromZividSettings<ConfigType>(defaultSettings);
+  server->dr_server.setConfigDefault(default_config);
+  server->dr_server.updateConfig(default_config);
+
+  // Setup the callback. This will invoke the callback, ensuring that the locally cached
+  // copy of config is updated.
+  auto cb = [& s = *server](const ConfigType& config, uint32_t /*level*/) {
```

Comment:
Nice way of not having to duplicate setting up these dynamic reconfigure server callbacks 👍